src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/AvailabilityService.kt

```diff
@@ -3,21 +3,91 @@ package uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service
 import org.slf4j.LoggerFactory
 import org.springframework.stereotype.Service
 import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.client.locationsinsideprison.LocationsInsidePrisonClient
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.client.nomismapping.NomisMappingClient
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.client.prisonapi.PrisonApiClient
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.entity.AppointmentSlot
 import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.model.request.AvailabilityRequest
 import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.model.response.AvailabilityResponse
 import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.repository.VideoAppointmentRepository
+import java.time.LocalDate
+import java.time.LocalTime
+import java.util.UUID
 
 @Service
 class AvailabilityService(
   private val videoAppointmentRepository: VideoAppointmentRepository,
   private val locationsInsidePrisonClient: LocationsInsidePrisonClient,
   private val availabilityFinderService: AvailabilityFinderService,
+  private val externalAppointmentsService: ExternalAppointmentsService,
 ) {
 
   companion object {
     private val log = LoggerFactory.getLogger(this::class.java)
   }
 
+  /**
+   * Assumptions:
+   *  - current booking journeys for BVLS allows only one person at one prison per booking.
+   *  - we consider ONLY the prison locations in the request for alternative times (we do not consider others - yet)
+   *  - the decision about whether a user is allowed to book into a locations is still the users own.
+   *  - we assume all appointments for this booking are on the same day and at the same prison
+   *
+   * To consider later:
+   *  - Get other VCC-enabled rooms at these prison(s) and offer them?
+   *  - Incorporate prison room decoration logic here, to find other rooms?
+   */
+  fun checkAvailability(request: AvailabilityRequest, includeExternalAppointments: Boolean = false): AvailabilityResponse {
+    if (!includeExternalAppointments) {
```

Comment:
Two functions with same name, and potentially same parameter signature, as the boolean is defaulted.